charts/invenio/templates/_helpers.tpl

```diff
@@ -167,42 +183,47 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
   {{- end }}
 {{- end -}}
 
-##########################     RabbitMQ vhost     ##########################
 {{/*
   This template renders the vhost for RabbitMQ.
 */}}
 {{- define "invenio.rabbitmq.vhost" -}}
   {{- if .Values.rabbitmq.enabled }}
     {{- "" }}
   {{- else }}
-    {{- required "Missing .Values.rabbitmqExternal.vhost" .Values.rabbitmqExternal.vhost }}
+    {{- required "Missing .Values.rabbitmqExternal.vhost" (tpl .Values.rabbitmqExternal.vhost .) }}
   {{- end }}
 {{- end -}}
 
-##########################     Celery broker URI     ##########################
 {{/*
-  This template renders the URI for connecting to RabbitMQ.
+  RabbitMQ connection env section. 
 */}}
-{{- define "invenio.rabbitmq.uri" -}}
-  {{- $username := (include "invenio.rabbitmq.username" .) -}}
-  {{- $password := (include "invenio.rabbitmq.password" .) -}}
-  {{- $port := (include "invenio.rabbitmq.amqpPort" .) -}}
-  {{- $hostname := (include "invenio.rabbitmq.hostname" .) -}}
-  {{- $protocol := (include "invenio.rabbitmq.protocol" .) -}}
-  {{- $vhost := (include "invenio.rabbitmq.vhost" .) -}}
-  {{- printf "%s://%s:%s@%s:%v/%s" $protocol $username $password $hostname $port $vhost}}
-{{- end -}}
-
-###########################     RabbitMQ API URI     ###########################
-{{/*
-  This template renders the URI for RabbitMQ's API endpoint.
-*/}}
-{{- define "invenio.rabbitmq.apiUri" -}}
-  {{- $username := (include "invenio.rabbitmq.username" .) -}}
-  {{- $password := (include "invenio.rabbitmq.password" .) -}}
-  {{- $port := (include "invenio.rabbitmq.managementPort" .) -}}
-  {{- $hostname := (include "invenio.rabbitmq.hostname" .) -}}
-  {{- printf "http://%s:%s@%s:%v/api/" $username $password $hostname $port }}
+{{- define "invenio.config.queue" -}}
+{{- $uri := "$(INVENIO_AMQP_BROKER_PROTOCOL)://$(INVENIO_AMQP_BROKER_USER):$(INVENIO_AMQP_BROKER_PASSWORD)@$(INVENIO_AMQP_BROKER_HOST):$(INVENIO_AMQP_BROKER_PORT)/$(INVENIO_AMQP_BROKER_VHOST)" -}}
+- name: INVENIO_AMQP_BROKER_USER
+  value: {{ include "invenio.rabbitmq.username" . }}
+- name: INVENIO_AMQP_BROKER_HOST
+  value: {{ include "invenio.rabbitmq.hostname" . }}
+- name: INVENIO_AMQP_BROKER_PORT
+  value: {{ include "invenio.rabbitmq.amqpPort" . }}
+- name: INVENIO_AMQP_BROKER_VHOST
+  value: {{ include "invenio.rabbitmq.vhost" . }}
+- name: INVENIO_AMQP_BROKER_PROTOCOL
+  value: {{ include "invenio.rabbitmq.protocol" . }}
+- name: INVENIO_AMQP_BROKER_PASSWORD
+{{- if or (and .Values.rabbitmq.enabled .Values.rabbitmq.auth.password) .Values.rabbitmqExternal.password }}
+  value: {{ include "invenio.rabbitmq.password" .  | quote }}
+{{- else }}
+  valueFrom:
+    secretKeyRef:
+      name: {{ include "invenio.rabbitmq.secretName" .}}
+      key: {{ include "invenio.rabbitmq.secretKey" .}}
+{{- end }}
+- name: INVENIO_BROKER_URL
+  value: {{ $uri }}
+- name: INVENIO_CELERY_BROKER_URL
+  value: {{ $uri }}
```

Comment:
Since the rendered chart contains the full `$uri` string, this looks a bit messy in my opinion:

![image](https://github.com/user-attachments/assets/76a27738-f598-49f4-844c-6ff307905be7)

WDYT about using `$(INVENIO_BROKER_URL)` instead, to make it explicit that they are the same?

```suggestion
- name: INVENIO_BROKER_URL
  value: {{ $uri }}
- name: INVENIO_CELERY_BROKER_URL
  value: $(INVENIO_BROKER_URL)
```

Reply:
I think it makes sense ... https://github.com/inveniosoftware/invenio-celery/blob/master/invenio_celery/config.py#L17 😂

Reply:
Good argument! :D I'll see to it.